server/repository/repository.go

```diff
@@ -100,8 +100,11 @@ func (s *Server) ListApps(ctx context.Context, q *RepoAppsQuery) (*RepoAppsRespo
 
 	items := make([]*AppInfo, 0)
 
-	for i := range ksonnetRes.Items {
-		items = append(items, &AppInfo{Type: string(repository.AppSourceKsonnet), Path: ksonnetRes.Items[i]})
+	for _, i := range ksonnetRes.Items {
+		t, err := repository.IdentifyAppSourceTypeByAppPath(i)
```

Comment:
Making this call per app.yaml is *way* too expensive for the app list query. We need a different mechanism to identify ksonnet apps by file lists alone.

Reply:
@jessesuen have pushed a change to address this